lib/virt_autotest/utils.pm

```diff
@@ -567,4 +568,29 @@ EOF
     return;
 }
 
+=head2 check_ssh_port_open
+
+Check whether ssh open tcp:22 is open on remote machine. This subroutine accepts two arguments,
+ssh_dst_machine and ssh_check_timeout, which are fqdn or ip address of remote machine and timeout 
+value respectively. Default timeout is 60s. ssh_dst_machine has no default value and test will 
+die if the subroutine is called without being passed value to ssh_dst_machine.
+
+=cut
+sub check_ssh_port_open {
+    my ($ssh_dst_machine, $ssh_check_timeout) = @_;
+    $ssh_dst_machine   //= "";
+    $ssh_check_timeout //= 60;
+    croak("IP address or FQDN should be given to argument ssh_dst_machine.") if ($ssh_dst_machine eq "");
+
+    my %ssh_port_hash = (
+        tcp => {
+            22 => {
+                name => 'SSH'
+            }
+        }
+    );
+
+    check_ports($ssh_dst_machine, $ssh_check_timeout, \%ssh_port_hash);
```

Comment:
[g](https://github.com/os-autoinst/os-autoinst-distri-opensuse/blob/master/lib/opensusebasetest.pm#L841) looks like a simple enough solution without needing the additional dependency. Maybe this is something you can consider

Reply:
I do not think I can reuse that one. I implemented a reusable function check_ssh_port_open which serves a single purpose

Reply:
You don't need to reuse the same code but just use the same perl module to do the port check

Reply:
I think I still prefer to use IO::Socket::PortState:check_ports. It should be OK, right ? I do not have to use IO::Socket::INET.

Reply:
What is the benefit of IO::Socket::PortState over IO::Socket::INET?

Reply:
I think it is a higher level wrapper of IO::Socket::INET. So you just use check_ports instead of "create new object".

Reply:
Updated.